docs/reference/mapping/params/format.asciidoc

```diff
@@ -31,8 +31,38 @@ down to the nearest day.
 [[custom-date-formats]]
 ==== Custom date formats
 
-Completely customizable date formats are supported. The syntax for these is explained
-https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html[DateTimeFormatter docs].
+Completely customizable date formats are supported. The syntax for these is explained in
+https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/time/format/DateTimeFormatter.html[DateTimeFormatter docs].
+
+[[custom-date-format-locales]]
+===== Differences in locale information between JDK versions
+
+There can be some differences in date formats between JDK versions and different locales. In particular,
+there can be differences in text strings used for textual date formats, and there can be differences
+in the results of week-date calculations.
+
+There can be differences in text strings used by the following field specifiers:
+
+* `B`, `E`, `G`, `O`, `a`, `v`, `z` of any length
+* `L`, `M`, `Q`, `q`, `c`, `e` of length 3 or greater
+* `Z` of length 4
+
+If the text format changes between Elasticsearch or JDK versions, it can cause significant problems
+with ingest, output, and re-indexing. It is recommended to always use numerical fields in custom date formats,
+which are not affected by locale information.
+
+There can also be differences in week-date calculations using the `Y`, `W`, and `w` field specifiers.
```

Comment:
I love how explicit the list of field specifiers is.

Would it make sense to word this advice such that we could add these three stragglers to the bullet list above, to provide one-stop-shopping for users who want to ensure they avoid this problem? I could envision a user swiping the above list and pasting it into a Jira ticket, thereby missing these three.

Reply:
Possibly, however the effects are quite different for each type, and I expect users to read the whole doc